opensrp-unicef-tunisia/src/main/java/org/smartregister/uniceftunisia/application/UnicefTunisiaApplication.java

```diff
@@ -264,65 +240,50 @@ public void onCreate() {
+                BuildConfig.VERSION_CODE, BuildConfig.DATABASE_VERSION);
         ImmunizationLibrary.getInstance().setVaccineSyncTime(3, TimeUnit.MINUTES);
         fixHardcodedVaccineConfiguration();
 
         ConfigurableViewsLibrary.init(context);
+
         ChildLibrary.init(context, getRepository(), getMetadata(), BuildConfig.VERSION_CODE, BuildConfig.DATABASE_VERSION);
-        // Init Reporting library
+        ChildLibrary.getInstance().setApplicationVersionName(BuildConfig.VERSION_NAME);
+        ChildLibrary.getInstance().setClientProcessorForJava(getClientProcessor());
+        ChildLibrary.getInstance().getProperties().setProperty(ChildAppProperties.KEY.FEATURE_SCAN_QR_ENABLED, "true");
+
         ReportingLibrary.init(context, getRepository(), null, BuildConfig.VERSION_CODE, BuildConfig.DATABASE_VERSION);
         ReportingLibrary.getInstance().addMultiResultProcessor(new TripleResultProcessor());
 
-        ActivityConfiguration activityConfiguration = new ActivityConfiguration();
-        activityConfiguration.setHomeRegisterActivityClass(AncRegisterActivity.class);
-        activityConfiguration.setLandingPageActivityClass(ChildRegisterActivity.class);
-        AncMetadata ancMetadata = new AncMetadata();
-        ancMetadata.setLocationLevels(GizUtils.getLocationLevels());
-        ancMetadata.setHealthFacilityLevels(GizUtils.getHealthFacilityLevels());
-        ancMetadata.setFieldsWithLocationHierarchy(Arrays.asList("village"));
-        AncLibrary.init(context, BuildConfig.DATABASE_VERSION, activityConfiguration, null, new GizAncRegisterQueryProvider(), ancMetadata);
-
-        OpdMetadata opdMetadata = new OpdMetadata(OpdConstants.JSON_FORM_KEY.NAME, OpdDbConstants.KEY.TABLE,
-                OpdConstants.EventType.OPD_REGISTRATION, OpdConstants.EventType.UPDATE_OPD_REGISTRATION,
-                OpdConstants.CONFIG, OpdFormActivity.class, BaseOpdProfileActivity.class, true);
-
-        OpdConfiguration opdConfiguration = new OpdConfiguration.Builder(OpdRegisterQueryProvider.class)
-                .setOpdMetadata(opdMetadata)
-                .setOpdRegisterProviderMetadata(GizOpdRegisterProviderMetadata.class)
-                .setOpdRegisterRowOptions(GizOpdRegisterRowOptions.class)
-                .setOpdRegisterSwitcher(GizOpdRegisterSwitcher.class)
-                .build();
-
-        OpdLibrary.init(context, getRepository(), opdConfiguration, BuildConfig.VERSION_CODE, BuildConfig.DATABASE_VERSION);
-
         Fabric.with(this, new Crashlytics.Builder().core(new CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG).build()).build());
 
         initRepositories();
         initOfflineSchedules();
 
         SyncStatusBroadcastReceiver.init(this);
-        LocationHelper.init(GizUtils.ALLOWED_LEVELS, GizUtils.DEFAULT_LOCATION_LEVEL);
+        LocationHelper.init(new ArrayList<>(Arrays.asList(BuildConfig.LOCATION_LEVELS)), BuildConfig.DEFAULT_LOCATION);
```

Comment:
Are these arguments right?  `LocationHelper.init(ArrayList<String> allowedLevels, String defaultLocationLevel)`

Reply:
Yea I extracted this to BuildConfig instead of hardcoding them in String Constants. WE also use different location levels.